src/System.Text.Primitives/System/Text/InvariantParser_ulong.cs

```diff
@@ -30,20 +30,23 @@ public static bool TryParse(Utf8String utf8Text, out ulong value, out int bytesC
                         return true;
                     }
                 }
-                try
+                else if (value > UInt64.MaxValue / 10) // overflow
                 {
-                    ulong candidate = checked(value * 10 + nextByte - '0');
```

Comment:
You might want to pull out this `nextByte - '0'` expression into a local for clarity. You could probably re-use it above in this check as well: `if (nextByte < '0' || nextByte > '9')`
i.e.

```C#
var nextByteVal = nextByte - '0';
if (nextByteVal < 9) // Values below '0' will wrap from the subtraction.
   ...
```